space_etl.py

```diff
@@ -22,25 +22,123 @@ def create_schema(self,schema_name):
     #         return cur
     
     def execute_query(self,query):
-        try:
-            with self.engine.connect() as conn:
-                result = conn.execute(text(query))
-                conn.commit()
-                return result
-        except Exception as e:
-            print(f"Error executing SQL query: {e}")
-            return None
-
+        with self.engine.connect() as conn:
+            result = conn.execute(text(query))
+            conn.commit()
+            return result
+        
     def write_dataframe(self,df,details):
         table_name = details['table_name']
-        schema_name = details['schema_name']     
-        # check if table exists
-        # if exists add column
-        # then load
-        #if not exists columns
-        # then load
+        schema_name = details['schema_name'] 
+        schema_handle = SchemaDriftHandle(db_name=self.db_name)
+        columns_to_add = schema_handle.check_schema_drift(df=df,details=details)
+        if columns_to_add:
+            schema_handle.handle_schema_drift(df=df,details=details,columns_to_add=columns_to_add)
+
+
+        # if isinstance(self,SchemaDriftHandle):
+        #     print("Performing Schema Drift Handle by adding necessary columns in the table...\n")
+        #     schema_handle = SchemaDriftHandle(self.db_name)
+        #     columns_to_add = schema_handle.check_schema_drift(df=df,details=details)
+        #     if columns_to_add:
+        #         schema_handle.handle_schema_drift(df=df,details=details,columns_to_add=columns_to_add)
+            
```

Comment:
Does the `write_dataframe` function checks if table exists or not before it checks case for `schema_drift`?
Also, does this function creates schema if the schema does not exists?